pgdrive/scene_creator/highway_vehicle/controller.py

```diff
@@ -209,99 +216,8 @@ def predict_trajectory_constant_speed(self, times: np.ndarray) -> Tuple[List[np.
         return tuple(
             zip(
                 *[
-                    self.scene.network.position_heading_along_route(route, coordinates[0] + self.speed * t, 0)
-                    for t in times
+                    self.traffic_mgr.map.road_network.
+                    position_heading_along_route(route, coordinates[0] + self.speed * t, 0) for t in times
```

Comment:
WTF, do we really need to call so deeply? I suggest to provide some API in the intermediate layer.

Reply:
This part will be abandoned in the future. We will design a new zombie vehicle system to replace the kinematic models of highwayEnv. The intermediate interface is actually the PGTrafficVehicle, which will make decision, do planning and control the zombie vehicle in the future.

Reply:
Good. Create a issue to record your idea?